sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -196,6 +254,40 @@ case class RLike(left: Expression, right: Expression)
   override def escape(v: String): String = v
   override def matches(regex: Pattern, str: String): Boolean = regex.matcher(str).find(0)
   override def toString: String = s"$left RLIKE $right"
+
+  override protected def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    val patternClass = classOf[Pattern].getName
+
+    val literalRight: String = right match {
+      case x @ Literal(value: String, StringType) => escape(value)
+      case _ => null
+    }
+
+    val leftGen = left.gen(ctx)
+    val rightGen = right.gen(ctx)
+
+    val patternCode =
+      if (literalRight != null) {
+        s"${patternClass} pattern = $patternClass.compile($literalRight);"
+      } else {
+        s"""
+          ${patternClass} pattern = ${patternClass}.compile(rightStr);
+        """
+      }
+
+    s"""
+      ${leftGen.code}
+      ${rightGen.code}
```

Comment:
Please use a logic like this:
````
codeA
nullCheckA
  codeB
  nullcheckB
````
This allows to skip the evaluation of `right` if `left` is already null.